src/RadMarshakAsymptoticCorr/test_radiation_marshak_asymptotic_corr.cpp

```diff
@@ -0,0 +1,347 @@
+	// predictor-corrector method [2] similar to SDC is REQUIRED for a correct solution.
+	//
+	// For discussion of the asymptotic-preserving property, see [1] and [2]. For
+	// a discussion of the exact, self-similar solution to this problem, see [3].
+	// Note that when used with an SDC time integrator, PLM (w/ asymptotic correction
+	// in Riemann solver) does a good job, but not quite as good as linear DG on this
+	// problem. There are some 'stair-stepping' artifacts that appear with PLM at low
+	// resolution that do not appear when using DG. This is likely the "wide stencil"
+	// issue discussed in [4].
+	//
+	// 1. R.G. McClarren, R.B. Lowrie, The effects of slope limiting on asymptotic-preserving
+	//     numerical methods for hyperbolic conservation laws, Journal of
+	//     Computational Physics 227 (2008) 9711–9726.
+	// 2. R.G. McClarren, T.M. Evans, R.B. Lowrie, J.D. Densmore, Semi-implicit time integration
+	//     for PN thermal radiative transfer, Journal of Computational Physics 227
+	//     (2008) 7561-7586.
+	// 3. Y. Zel'dovich, Y. Raizer, Physics of Shock Waves and High-Temperature Hydrodynamic
+	//     Phenomena (1964), Ch. X.: Thermal Waves.
+	// 4. Lowrie, R. B. and Morel, J. E., Issues with high-resolution Godunov methods for
+	//     radiation hydrodynamics, Journal of Quantitative Spectroscopy and
+	//     Radiative Transfer, 69, 475–489, 2001.
+
+	// Problem parameters
+	const int max_timesteps = 1e6;
+	const double CFL_number = 10.0;
+	const double initial_dt = 5.0e-12; // s
+	const double max_dt = 5.0;	   // s
+	const double max_time = 10.0e-9;   // s
+	// const int nx = 60; // [18 == matches resolution of McClarren & Lowrie (2008)]
+	// const double Lx = 0.66; // cm
+
+	// Problem initialization
+	std::cout << "radiation constant (code units) = " << RadSystem_Traits<SuOlsonProblemCgs>::radiation_constant << "\n";
+	std::cout << "c_light (code units) = " << RadSystem_Traits<SuOlsonProblemCgs>::c_light << "\n";
+	std::cout << "rho * c_v = " << rho0 * c_v << "\n";
+	std::cout << "initial_dt = " << initial_dt << "\n";
+	std::cout << "max_dt = " << max_dt << "\n";
+	std::cout << "max_time = " << max_time << std::endl;
+
+	constexpr int nvars = RadSystem<SuOlsonProblemCgs>::nvar_;
+	amrex::Vector<amrex::BCRec> BCs_cc(nvars);
+	for (int n = 0; n < nvars; ++n) {
+		BCs_cc[n].setLo(0,
+				amrex::BCType::ext_dir);     // custom (Marshak) x1
+		BCs_cc[n].setHi(0, amrex::BCType::foextrap); // extrapolate x1
+		for (int i = 1; i < AMREX_SPACEDIM; ++i) {
+			BCs_cc[n].setLo(i, amrex::BCType::int_dir); // periodic
+			BCs_cc[n].setHi(i, amrex::BCType::int_dir);
+		}
+	}
+
+	// Problem initialization
+	RadhydroSimulation<SuOlsonProblemCgs> sim(BCs_cc);
+
+	sim.radiationReconstructionOrder_ = 3; // PPM
+	sim.stopTime_ = max_time;
+	sim.initDt_ = initial_dt;
+	sim.maxDt_ = max_dt;
+	sim.radiationCflNumber_ = CFL_number;
+	sim.maxTimesteps_ = max_timesteps;
+	sim.plotfileInterval_ = -1;
+
+	sim.use_wavespeed_correction_ = true;
```

Comment:
You could read this from the input file, e.g.:
```
amrex::ParmParse pp("marshak");
pp.query("use_wavespeed_correction", sim.use_wavespeed_correction_);
```
Then you could re-use the same problem file, and just create two tests (using different input files).

Reply:
Done.